megatron/core/inference/text_generation_controllers/text_generation_controller.py

```diff
@@ -662,6 +662,10 @@ def _dynamic_step_sample_logits(self, logits: Tensor, backend: str = "torch") ->
         context = self.inference_wrapped_model.inference_context
         materialize_only_last_token_logits = context.materialize_only_last_token_logits
 
+        # We do not want to sample tokens beyond vocabulary size.
```

Comment:
`logits` is a tensor created under inference mode. Would it not be safer to use `torch.where` to avoid an in-place update error? e.g.

```
mask = torch.arange(logits.size(-1), device=logits.device)[None, None, :]
        logits = torch.where(
            mask < self.tokenizer.vocab_size,
            logits,
            -torch.inf
        )
```